src/qibolab/instruments/zhinst.py

```diff
@@ -288,7 +288,7 @@ class Zurich(Controller):
 
     PortType = ZhPort
 
-    def __init__(self, name, descriptor, use_emulation=False, time_of_flight=0.0, smearing=0.0):
+    def __init__(self, name, descriptor=None, device_setup=None, use_emulation=False, time_of_flight=0.0, smearing=0.0):
```

Comment:
**suggestion:** Instead of adding optional arguments, alternate constructors might make this interface cleaner. I suggest adding two class methods:

``` py
def from_descriptor(cls, descriptor: ?, name: str, ...):
    ...
```

``` py
def from_device_setup(cls, device_setup: DeviceSetup, name: str, ...):
    ...
```

Reply:
Yeah, my intention was only to temporarily keep backwards compatibility.
We should probably just drop `descriptor` from the constructor (removing default `None` from `device_setup`), but I'd like to discuss it with @stavros11 and @Jacfomg.